### PR TITLE
feat: Add OP cli flag to opt-in into discv4 discovery

### PR DESCRIPTION
**Pull request by nkysg**
towards https://github.com/paradigmxyz/reth/issues/9919

### CODE REVIEW

File: crates/optimism/node/src/node.rs [hunk 1]
```diff
@@ -286,12 +291,15 @@ where
...
 
         let args = &ctx.config().network;
 
-        let network_builder = ctx
-            .network_config_builder()?
+        let mut network_config_builder = ctx.network_config_builder()?;
+        if disable_discovery_v4 {
+            network_config_builder = network_config_builder.disable_discv4_discovery();
+        }
+        let network_builder = network_config_builder
```
Comment (Rjected) [inline crates/optimism/node/src/node.rs:302]: should this be:
```suggestion
        if disable_discovery_v4 || args.discovery.disable_discovery {
            network_config_builder = network_config_builder.disable_discv4_discovery();
        }
        let network_builder = network_config_builder
```
or rather, what is the expected behavior when both the `--disable-discovery` and `--rollup.discovery.v4` flags are provided?

**Review by Rjected — COMMENTED** [review]
just want to confirm what we want w.r.t ux

**Review by Rjected — APPROVED** [review]
lgtm, thanks!

File: crates/optimism/node/src/node.rs [hunk 2]
```diff
@@ -286,12 +291,15 @@ where
...
 
         let args = &ctx.config().network;
 
-        let network_builder = ctx
-            .network_config_builder()?
+        let mut network_config_builder = ctx.network_config_builder()?;
+        if disable_discovery_v4 || args.discovery.disable_discovery {
+            network_config_builder = network_config_builder.disable_discv4_discovery();
+        }
+        let network_builder = network_config_builder
```
Comment (mattsse) [inline crates/optimism/node/src/node.rs:302]: style nit, you can use the `apply` function to keep the builder style chain and perform this check in the closure

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]

**Pull request merged**